tests/semanticdb/metac.expect

```diff
@@ -3808,13 +3808,13 @@ flags/p/package.Y# => final class Y
 flags/p/package.Y#`<init>`(). => primary ctor <init>
 flags/p/package.Z# => sealed trait Z
 flags/p/package.Z#`<init>`(). => primary ctor <init>
-flags/p/package.`y_=`(). => var method y_=
+flags/p/package.`y_=`(). => var method protected y_=
 flags/p/package.`y_=`().(x$1) => param x$1
 flags/p/package.m(). => macro m
 flags/p/package.m().[TT] => typeparam TT
-flags/p/package.x. => lazy val method x
+flags/p/package.x. => lazy val method protected[flags/p/] x
```

Comment:
I'm not sure this is right, in scala 2 semanticdb the access is printed as `private`, and the definition in  `tests/semanticdb/expect/semanticdb-Flags.scala` is `private lazy val x = 1`

Reply:
to debug you can try using this script I made, save it to `compiler/test/dotty/tools/DottySymbolStealer.scala`

If I run it with
```
scala3-compiler-bootstrapped/Test/runMain dotty.tools.printSymFlags "package object flags { private lazy val x = 1 }" "x"
```

I get the output 
```
val x => lazy <touched> private[module class flags]
```

```scala
package dotty.tools

import dotc.ast.tpd
import dotc.core.Names._
import dotc.ast.tpd._
import dotc.core.Contexts.Context
import dotc.core.Symbols.Symbol
import dotc.core.Decorators._
import dotc.core.Types.Type

@main def printSymFlags(source: String, termRefs: String*) = {
  val (ctx, syms) = DottySymbolStealer.stealSymbol(source, termRefs*)
  given Context = ctx
  syms.map(sym =>
    val priv = sym.privateWithin
    s"$sym => ${sym.flags.flagStrings(if priv.exists then s"$priv" else "").mkString(" ")}"
  ).foreach(println)
}

object DottySymbolStealer extends DottyTest {
  def stealSymbol(source: String, termRefs: String*): (Context, List[Symbol]) = {
    var scontext : Context = null
    var tp: List[Symbol] = null
    checkCompile("typer", source) {
      (tree, context) =>
        given Context = context
        val remaining = scala.collection.mutable.Set.from(termRefs)
        val findValDef: (List[ValDef], tpd.Tree) => List[ValDef] =
          (acc , tree) =>  tree match {
            case t: ValDef =>
              remaining.find(t.name.toString == _) match
                case Some(name) =>
                  remaining -= name
                  t :: acc
                case _ =>
                  acc
            case _ => acc
          }
        val d = new DeepFolder[List[ValDef]](findValDef).foldOver(Nil, tree)
        tp = d.map(_.symbol).reverse
        scontext = context
    }
    (scontext, tp)
  }
}
```

Reply:
so using scala 2 repl it appears that private values in the package object do not get a `privateWithin` symbol, but I think its ok to keep it in scala 3